### PR TITLE
Make update work for Darwin, Linux, and Windows(powershell)

### PR DESCRIPTION
**Pull request by dcwangmit01**
* Added Linux Support
* Consolidate Windows support to powershell only.
* Windows Git bash is no longer supported
* Made binary update itself in place
* Removed all assumptions on binary location.
* Added check to ensure sudo required for linux and darwin

### CODE REVIEW

File: pkg/kdk/update.go [hunk 1]
```diff
@@ -41,6 +42,11 @@ func WarnIfUpdateAvailable(cfg *KdkEnvConfig) {
 		return
 	}
 
+	// Add a sudo hint for unix-based OS's
+	sudo := ""
+	if runtime.GOOS == "linux" || runtime.GOOS == "darwin" {
+		sudo = "sudo "
```
Comment (rtluckie) [inline pkg/kdk/update.go]: trailing space intentional?

File: pkg/kdk/update.go [hunk 2]
```diff
@@ -135,97 +145,96 @@ func Update(cfg *KdkEnvConfig) {
...
-
-	// Figure out the binary path
-	//   TODO: Don't make assumptions on binary path, like /usr/local/bin
-	kdkBinName := "kdk"
-	if runtime.GOOS == "windows" {
-		kdkBinName = kdkBinName + ".exe"
-	}
-	// TODO: Pass individual segments into Join
-	kdkBinPath := filepath.Join("/usr/local/bin", kdkBinName)
+	///////////////////////////////////////
```
Comment (rtluckie) [inline pkg/kdk/update.go]: need to remove ^

File: pkg/kdk/update.go (hunk 1 above)
Comment (dcwangmit01) [reply pkg/kdk/update.go]: Yes, space is intentional.  Added comment

File: pkg/kdk/update.go (hunk 2 above)
Comment (dcwangmit01) [reply pkg/kdk/update.go]: Done

File: pkg/kdk/update.go [hunk 3]
```diff
@@ -112,13 +121,13 @@ func Update(cfg *KdkEnvConfig) {
 	}
 
 	if needsUpdateImage(cfg) {
-		log.Info("Updating KDK image")
-		err := pullImage(cfg, cfg.ConfigFile.AppConfig.ImageRepository + ":" + latestReleaseVersion)
+		log.Info("Updating KDK docker image")
```
Comment (josdotso) [inline pkg/kdk/update.go]: MAYBE: better to say "KDK container" instead of "docker".

File: pkg/kdk/update.go [hunk 4]
```diff
@@ -112,13 +121,13 @@ func Update(cfg *KdkEnvConfig) {
...
+		err := pullImage(cfg, cfg.ConfigFile.AppConfig.ImageRepository+":"+latestReleaseVersion)
```
Comment (josdotso) [inline pkg/kdk/update.go]: Why dropped the spaces here?  It's harder to read without them.

Comment (dcwangmit01) [reply pkg/kdk/update.go]: gofmt wants to drop the whitespace.

**Pull request merged**